### ISSUE TITLE
MediaRecorder prepare failed on Android

### ISSUE THREAD

**Issue by kenle** [Isvisoft/flutter_screen_recording#13]
Hi, I'm having trouble getting this plugin to work on Android devices.  

I get the error below, I believe related to the output file. I'm no Android expert, and hours of search on Google about MediaRecorder, getExternalStoragePublicDirectory, and setOutpuFile gives me no clearer clue as to why this is happening.  So just want to ask, anything come to mind why this error may occur?

<img width="867" alt="Screen Shot 2020-04-02 at 02 11 16" src="https://user-images.githubusercontent.com/47367/78176919-7cb70e80-7487-11ea-8a15-a3714e482af0.png">

My device information: 
Samsung Galaxy Tab A (2019)
Android 9 (API 28)

I've also used permission_handler plugin to request permission for external storage.

Thank you very much.

**Comment by kenle** [comment]
It is resolved.  In case anyone else runs into problems with Android, I give a quick run through of the modifications I made to get it working.  Sorry these are screenshots and not links to the actual code lines.

The following modifications were made in file FlutterScreenRecordingPlugin.kt

1. The original mDisplayWidth and mDisplayHeight were causing MediaRecorder to fail for some reason, I modified as follows.  
![image](https://user-images.githubusercontent.com/47367/78257158-ad4c8600-7523-11ea-9f46-962be5117b17.png)

2. I found I had to edit Activity.Compat.startActivityForResult to use registrar.activity(), else I was getting a null exception error with the current code.

![image](https://user-images.githubusercontent.com/47367/78257272-d53be980-7523-11ea-81ea-e6581b1e425b.png)

3. Finally, I had to modify declaration of late init variable _result, else I was also getting a null exception error.

![image](https://user-images.githubusercontent.com/47367/78257368-f43a7b80-7523-11ea-9223-babfa8449f91.png)
